src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -257,23 +257,35 @@ public Task AnnounceBlocks(List<uint256> blockHashesToAnnounce)
                     {
                         headers.Add(chainedBlock.Header);
                     }
-                    else if (chainBehavior.PendingTip.GetAncestor(chainedBlock.Height) != null)
```

Comment:
I would suggest an improvement, this code can move the the state changed even and assign to ta local variable (see the con manager behavior)
` var chainBehavior = node.Behavior<ChainHeadersBehavior>();`

Reply:
i don't understand what you mean